src/es/mangaesp/src/eu/kanade/tachiyomi/extension/es/mangaesp/MangaEsp.kt

```diff
@@ -0,0 +1,264 @@
+    ): Observable<MangasPage> {
+        return if (comicsList.isEmpty()) {
+            client.newCall(searchMangaRequest(page, query, filters))
+                .asObservableSuccess()
+                .map { searchMangaParse(it, page, query, filters) }
+        } else {
+            Observable.just(parseComicsList(page, query, filters))
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request = GET("$baseUrl/api/comics", headers)
+
+    override fun searchMangaParse(response: Response): MangasPage = throw UnsupportedOperationException()
+
+    private fun searchMangaParse(response: Response, page: Int, query: String, filters: FilterList): MangasPage {
+        val responseData = json.decodeFromString<ComicsDto>(response.body.string())
+        comicsList = responseData.response.toMutableList()
+        return parseComicsList(page, query, filters)
+    }
+
+    private var filteredList = mutableListOf<SeriesDto>()
+
+    private fun parseComicsList(page: Int, query: String, filterList: FilterList): MangasPage {
+        if (page == 1) {
+            filteredList.clear()
+            
+            if (query.isNotBlank()) {
+                if (query.length < 2) throw Exception("La búsqueda debe tener al menos 2 caracteres")
+                filteredList.addAll(comicsList.filter { it.name.contains(query, ignoreCase = true) })
+            } else {
+                filteredList.addAll(comicsList)
+            }
+
+            val statusFilter = filterList.firstInstanceOrNull<StatusFilter>()
+
+            if (statusFilter != null) {
+                filteredList = filteredList.filter { it.status == statusFilter.toUriPart() }.toMutableList()
+            }
+
+            val sortByFilter = filterList.firstInstanceOrNull<SortByFilter>()
+
+            if (sortByFilter != null) {
+                if (sortByFilter.state?.ascending == true) {
+                    when (sortByFilter.selected) {
+                        "name" -> filteredList.sortBy { it.name }
+                        "views" -> filteredList.sortBy { it.trending?.views }
+                        "updated_at" -> filteredList.sortBy { it.lastChapterDate }
+                        "created_at" -> filteredList.sortBy { it.createdAt }
+                    }
+                } else {
+                    when (sortByFilter.selected) {
+                        "name" -> filteredList.sortByDescending { it.name }
+                        "views" -> filteredList.sortByDescending { it.trending?.views }
+                        "updated_at" -> filteredList.sortByDescending { it.lastChapterDate }
+                        "created_at" -> filteredList.sortByDescending { it.createdAt }
+                    }
+                }
+            }
+        }
+
+        val hasNextPage = filteredList.size > page * MANGAS_PER_PAGE
+
+        return MangasPage(
+            filteredList.subList((page - 1) * MANGAS_PER_PAGE, filteredList.size)
```

Comment:
wouldn't this just return the original list back? probably need to do something like:
 
```suggestion
            filteredList.subList((page - 1) * MANGAS_PER_PAGE, min(page * MANGAS_PER_PAGE, filteredList.size)
```

Reply:
true, my bad